x-pack/plugins/cases/public/mocks.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CasesUiStart } from './types';
+
+const createStartContract = (): jest.Mocked<CasesUiStart> => ({
+  getAllCases: jest.fn(),
+  getAllCasesSelectorModal: jest.fn(),
+  getCaseView: jest.fn(),
+  getConfigureCases: jest.fn(),
+  getCreateCase: jest.fn(),
+  getRecentCases: jest.fn(),
+});
+
+export const casesPluginMock = {
```

Comment:
where is this used?

Reply:
I think is good to have them if any other plugin wants to use our mocks as we do with core for example.

Reply:
Currently it's used here: https://github.com/jonathan-buttner/kibana/blob/cases-rbac-bugs/x-pack/plugins/security_solution/public/overview/components/sidebar/sidebar.test.tsx#L25

It seemed like a common pattern to expose a mock for the object we return from the plugin's lifecycle. Here are a few  examples:
https://github.com/elastic/kibana/blob/master/x-pack/plugins/security/public/mocks.ts
https://github.com/elastic/kibana/blob/master/x-pack/plugins/reporting/public/mocks.ts